WordPress/gutenberg
# [Mobile] - RichText - Parse CSS values and avoid setting undefined ones

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5391

This PR adds support for parsing CSS line-height values.

## What?
Some themes are including line-height values like `calc( 1em + 0.875rem )` in their `theme.json` configuration which currently breaks the editor in development mode, in production doesn't crash but it doesn't work as expected.

## Why?
To support current and future themes that include CSS vars or CSS expressions as line-height values.

## How?
First by updating the `getPxFromCssUnit` util which was currently breaking while trying to parse values like `calc( 1em + 0.875rem )` where it was splitting the `+` character and the decimals as well. Where it should only split math expression values:  `+`, `-`, `*`, etc.

Second by updating the `getLineHeight` method to parse these values.

It also includes a test in both the helper and RichText to take into account these cases.

## Testing Instructions

- Open WordPress iOS/Android
- Select the Masu theme
- Open the editor
- Tap on the empty placeholder
- Type a text
- **Expect** the font size to show correctly
- **Expect** the editor not to crash

**Repeat with a different theme.**

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
Before | After
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/212158096-c709bec9-f4f0-4605-888c-f2cd8dad306d.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/212158127-13010888-b18a-4ece-ade5-beb1def1de2b.png" width="200" /></kbd>
